packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -193,6 +194,63 @@ export function renderToInitialFizzStream({
   )
 }
 
+function createMetadataTransformStream(
+  insert: () => Promise<string> | string
+): TransformStream<Uint8Array, Uint8Array> {
+  let iconMarkIndex = -1
+  let closedBodyIndex = -1
+  let isMarkReplaced = false
+  let isScriptInserted = false
+
+  return new TransformStream({
+    async transform(chunk, controller) {
+      if (isScriptInserted) {
+        controller.enqueue(chunk)
+        return
+      }
+      // Only search for the closed head tag once
+      if (iconMarkIndex === -1) {
+        iconMarkIndex = indexOfUint8Array(chunk, ENCODED_TAGS.META.ICON_MARK)
+      }
+
+      if (iconMarkIndex !== -1 && !isMarkReplaced) {
+        // Remove the icon mark <meta name="«nxt-icon»" content="" />
+        // to noted that we need to add icon re-insertion script later.
+        // This should always be inserted
+        const replacedChunk = replaceInUint8Array(
+          chunk,
+          ENCODED_TAGS.META.ICON_MARK,
+          encoder.encode(''),
+          iconMarkIndex
+        )
+        chunk = replacedChunk
+        isMarkReplaced = true
+      }
+      if (closedBodyIndex === -1) {
+        closedBodyIndex = indexOfUint8Array(chunk, ENCODED_TAGS.CLOSED.BODY)
+      }
+      // Insert the re-insertion script tag at the end.
+      // There're be few cases:
+      // - When metadata is blocking, we don't have this transform and metadata stays in head.
+      // - When head is prerendered, metadata will be in the body.
```

Comment:
I'm still a bit confused by this case... if the head was prerenderable, why would it end up in the body?

Reply:
I updated the comment, it's like the case I mentioned in another PR. No matter metadata is static or dynamic, if it's suspended, it's likely ended up in `<body>` rather than in `<head>`.